docker-host/main.go

```diff
@@ -0,0 +1,164 @@
+		command = command + fmt.Sprintf(" --registry-mirror=%s", REGISTRY_MIRROR)
+	}
+	var cmd = exec.Command("sh", "-c", command)
+
+	if dockerHost != cli.DaemonHost() {
+		log.Fatalf("Could not connect to %s", dockerHost)
+	}
+	c := cron.New()
+	pruneImages(cli, c)
+	removeExited(cli, c)
+	updateImages(cli, c)
+	c.Start()
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		log.Fatalf("could not run command: ", err)
+	}
+}
+
+func pruneImages(client *client.Client, c *cron.Cron) {
+	c.AddFunc(pruneImagesSchedule, func() {
+		log.Println("Starting image prune")
+		ageFilter := filters.NewArgs()
+		pruneDanglingFilter := filters.NewArgs()
+		ageFilter.Add("until", pruneImagesUntil)
+		pruneDanglingFilter.Add("dangling", danglingFilter)
+
+		// # prune all images older than 7 days or what is specified in the environment variable
+		_, err := client.ImagesPrune(context.Background(), ageFilter)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		// # prune all docker build cache images older than 7 days or what is specified in the environment variable
+		_, buildErr := client.BuildCachePrune(context.Background(), types.BuildCachePruneOptions{Filters: ageFilter})
+		if buildErr != nil {
+			log.Println(buildErr)
+		}
+		// # after old images are pruned, clean up dangling images
+		_, pruneErr := client.ImagesPrune(context.Background(), pruneDanglingFilter)
+		if pruneErr != nil {
+			log.Println(pruneErr)
+		}
+		log.Println("Prune complete")
+	})
+}
+
+func removeExited(client *client.Client, c *cron.Cron) {
+	c.AddFunc(removeExitedSchedule, func() {
+		log.Println("Starting removeExited")
+		ctx := context.Background()
+		statusFilter := filters.NewArgs()
+		statusFilter.Add("status", "exited")
+		containers, err := client.ContainerList(ctx, types.ContainerListOptions{
+			Filters: statusFilter,
+		})
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		// # remove all exited containers
+		for _, container := range containers {
+			err := client.ContainerRemove(ctx, container.ID, types.ContainerRemoveOptions{
+				Force:         true,
+				RemoveVolumes: true,
+			})
+			if err != nil {
+				log.Println(err)
+			}
+		}
+		log.Println("removeExited complete")
+	})
+}
+
+func updateImages(client *client.Client, c *cron.Cron) {
+	c.AddFunc(updateImagesSchedule, func() {
+		log.Println("Starting update images")
+		ctx := context.Background()
+		filters := addFilters(repositoriesToUpdate)
+		images, err := client.ImageList(ctx, types.ImageListOptions{Filters: filters})
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		var imgRepoTags []string
+		for _, img := range images {
+			imgRepoTags = append(imgRepoTags, img.RepoTags...)
+		}
+
+		// # Iterates through all images that have the name of the repository we are interested in in it
+		for _, image := range imgRepoTags {
+			out, err := client.ImagePull(ctx, image, types.ImagePullOptions{})
+			log.Println("Image to update", image)
```

Comment:
Just to be clear what it's doing in the logs

```suggestion
			log.Println("Checking update for", image)
```

Reply:
The only other point here is that it could be a bit noisy always listing all the times images are checked for updates - and instead, maybe we only log the ones we actually update on a given run (if that;s even possible), or we count how many images were checked?